### ISSUE TITLE
Can you change version v0 to v2020-12-01 for GetCatalogItem?

### ISSUE THREAD

**Issue by aispel** [amazon-php/sp-api-sdk#101]
https://sellercentral-japan.amazon.com/forums/t/topic/81380
2022/03 the v0 of GetCatalogItem will be abolished.
Can you change version v0 to v2020-12-01 for GetCatalogItem?

**Comment by norberttech** [comment]
Could you provide an English version of this announcement? Does it mean that they will deprecate the following endpoint? 

![image](https://user-images.githubusercontent.com/1921950/147390267-f10b6606-42eb-42fa-abf4-5293be8bf61d.png)

**Comment by aispel** [comment]
Sorry, I can not find the English version.

It means the two APIs( Catalog Items v0) as below will be deprecated.
Selling Partner API for Catalog Items v0
ListCatalogItems > Selling Partner API for Catalog Items v2020-12-01 SearchCatalogItems
GetCatalogItem > Selling Partner API for Catalog Items v2020-12-01 GetCatalogItem

We should use /catalog/2020-12-01/ instead of /catelog/v0
(https://github.com/amzn/selling-partner-api-docs/blob/main/references/catalog-items-api/catalogItems_2020-12-01.md)

**Comment by norberttech** [comment]
hmm I see, ok let me think about it for a while, I need to figure out the best way to introduce that change into this lib to not break apps that are update by dependabot

**Comment by norberttech** [comment]
hey @aispel, I just released branch 3.x that comes with v2020-12-01 version of CatalogItem API Model, 
it's now the default branch, for the time being, it will be maintained together with branch 2.x until v0 will get fully deprecated.
Closing this one

**Comment by aispel** [comment]
Thank you so much.
But I run the sample code, it shows the error like below.
For old version, it was ok.
Fatal error: Uncaught TypeError: Argument 4 passed to AmazonPHP\SellingPartner\Api\CatalogApi\CatalogItemSDK::getCatalogItem() must be of the type array, string given,

**Comment by norberttech** [comment]
this is why v2020-12-01 is considered as a BC break, please find changes in this method below: 
```diff
-public function getCatalogItem(AccessToken $accessToken, string $region, string $marketplace_id, string $asin) : \AmazonPHP\SellingPartner\Model\CatalogItem\GetCatalogItemResponse
+public function getCatalogItem(AccessToken $accessToken, string $region, string $asin, array $marketplace_ids, array $included_data = null, string $locale = null) : \AmazonPHP\SellingPartner\Model\CatalogItem\Item
```

`$marketplace_id` expects now an array of marketplace id's instead of single string

**Comment by aispel** [comment]
Thank you very much. I have got the item information.
But when I add 'attributes' to the included_data, it has an error. 

$item = $sdk->catalogItem()->getCatalogItem(
        $accessToken,
        Regions::NORTH_AMERICA,
        $asin = 'B08176SM7C',
        $marketplace_ids = [Marketplace::US()->id()],
        $included_data = ['images','summaries','salesRanks','productTypes','attributes'],
    );

Fatal error: Uncaught TypeError: Argument 1 passed to AmazonPHP\SellingPartner\Model\CatalogItem\Item::setAttributes() must be an object or null, array given, called in /Applications/MAMP/htdocs/yts/vendor/amazon-php/sp-api-sdk/src/AmazonPHP/SellingPartner/ObjectSerializer.php on line 376 and defined in /Applications/MAMP/htdocs/yts/vendor/amazon-php/sp-api-sdk/src/AmazonPHP/SellingPartner/Model/CatalogItem/Item.php:298 Stack trace: #0 /Applications/MAMP/htdocs/yts/vendor/amazon-php/sp-api-sdk/src/AmazonPHP/SellingPartner/ObjectSerializer.php(376): AmazonPHP\SellingPartner\Model\CatalogItem\Item->setAttributes(Array) #1 /Applications/MAMP/htdocs/yts/vendor/amazon-php/sp-api-sdk/src/AmazonPHP/SellingPartner/Api/CatalogApi/CatalogItemSDK.php(144): AmazonPHP\SellingPartner\ObjectSerializer::deserialize(Object(AmazonPHP\SellingPartner\Configuration), Object(stdClass), 'AmazonPHP\\Selli...', Array) #2

**Comment by norberttech** [comment]
This indeed looks like a bug, thanks for testing and reporting it! I will take a look into this one shortly, most likely it has something to do with deserialization

**Comment by norberttech** [comment]
ok, I think I know what's going on, [ItemAttribute](https://github.com/amzn/selling-partner-api-models/blob/main/models/catalog-items-api-model/catalogItems_2020-12-01.json#L1081) in the model file is defined as an "object" which is later translated by the open api generator into something like this: 
![image](https://user-images.githubusercontent.com/1921950/147581778-376535af-1373-4ab4-a650-bee6729fe7c0.png)

Now, when rector is used to add types based on `@param` instead of making an `array` is creating `object` type.

I need to think how to solve this one, unfortunately there are 3 more places defined as `object` without any properties so they are all mapped into `object`

**Comment by norberttech** [comment]
I think I found a decent way to fix this issue through rector. 
The same bug was also present on 2.x branch, I just released patches for both maintained branches, `3.0.1` and 2.1.1`

It should be all good now

**Comment by aispel** [comment]
Thank you very much. It is ok!
